### PR TITLE
Update libgit2 to v1.1.1

### PR DESCRIPTION
**Pull request by jairbubbles**
I'm trying to update libgit2 to a more recent version to get support for the sparse checkout (see https://github.com/libgit2/libgit2sharp/issues/1845)

### CODE REVIEW

**Review by A-Ovchinnikov-mx — COMMENTED** [review]
There seem to be some changes in the API in the recent versions of libgit2, so some more work is required. Please have a look at: https://github.com/mendix/libgit2sharp/commit/f958478f89fa37e80881a63a621b6a56712297ec

We've forked the project to incorporate some changes we need. Unfortunately, we could not get much attention for our pull requests in this repository, and we kept accidentally creating new pull requests to this upstream repo, so we had to disconnect our fork from it.

**Comment by jairbubbles** [comment]
Thanks for the info @A-Ovchinnikov-mx  ! It's too bad we can't all work on the upstream repository 😭

**Comment by ethomson** [comment]
You can.  https://github.com/libgit2/libgit2sharp/issues/1834

**Comment by A-Ovchinnikov-mx** [comment]
I would love to help but I'm a bit afraid I won't be able to dedicate too much time to it 😬 And that would be my personal investment, not from the company. If that's already good enough, then I can drop a message in that thread from my personal GitHub account.

**Comment by jairbubbles** [comment]
Hello @bording, I saw you did some modification recently and I was wondering what do you think of moving forward with the libgit version? A new version 1.1.1 was also published that we could target.

I looked at @A-Ovchinnikov-mx commits but the fixes need to be done on libgit2sharp side.

**Comment by bording** [comment]
> Hello @bording, I saw you did some modification recently and I was wondering what do you think of moving forward with the libgit version? A new version 1.1.1 was also published that we could target.

My plan is to move up to a newer version of libgit2, but I was hoping to wait for https://github.com/libgit2/libgit2/pull/5974 land first.

File: build.libgit2.ps1 [hunk 1]
```diff
@@ -93,8 +93,6 @@ function Assert-Consistent-Naming($expected, $path) {
     $dll = get-item $path
 
     Ensure-Property $expected $dll.Name "Name" $dll.Fullname
-    Ensure-Property $expected $dll.VersionInfo.InternalName "VersionInfo.InternalName" $dll.Fullname
```
Comment (bording) [inline build.libgit2.ps1:96]: Why were these removed?

Comment (jairbubbles) [reply build.libgit2.ps1:96]: I didn't investigate much but it was failing. I can look into it more of you want.

**Comment by jairbubbles** [comment]
I'm not particularly in a hurry so that's fine with me to wait for a newer version. In the meantime, would it be possible to publish an alpha version of the native binaries NuGet packages so that I can create a draft PR on libgit2sharp repo to make sure the CI is all green?

File: build.libgit2.ps1 (hunk 1 above)
Comment (bording) [reply build.libgit2.ps1:96]: Since those aren't failing in master, it would be good to know what the problem is.

Comment (jairbubbles) [reply build.libgit2.ps1]: So the regression comes from https://github.com/libgit2/libgit2/pull/5547. The define is not set anymore when compiling the `.rc` file.

If we replace the line:
`target_compile_definitions(git2internal PRIVATE LIBGIT2_FILENAME=\"${LIBGIT2_FILENAME}\")`
by 
`target_compile_definitions(git2 PRIVATE LIBGIT2_FILENAME=\"${LIBGIT2_FILENAME}\")`

It works correctly.

I'll create a PR, in parallel I'll try to find a workaround.

Comment (jairbubbles) [reply build.libgit2.ps1]: I finally ended up patching the file in the powershell script. 
The PR is there: https://github.com/libgit2/libgit2/pull/5994

**Review by bording — APPROVED** [review]
